lib/node_modules/@stdlib/utils/parse-ndjson/lib/main.js

```diff
@@ -0,0 +1,81 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
+var isFunction = require( '@stdlib/assert/is-function' );
+var parseJSON = require( '@stdlib/utils/parse-json' );
+
+
+// MAIN //
+
+/**
+* Parses a string as newline-delimited JSON (NDJSON).
+*
+* @param {string} str - input string
+* @param {Function} [reviver] - transformation function applied to each line
+* @throws {TypeError} first argument must be a string
+* @throws {TypeError} reviver must be a function
+* @returns {Array<(*|Error)>} array of parsed values or parse errors
```

Comment:
```suggestion
* @returns {Array|Error} array of parsed values or parse errors
```

The previous type annotation implies that an array of errors could be returned. That is incorrect, as the function early returns.